matrix_bot_api/matrix_bot_api.py

```diff
@@ -10,20 +10,24 @@ class MatrixBotAPI:
     # password - Matrix password
     # server   - Matrix server url : port
     # rooms    - List of rooms ids to operate in, or None to accept all rooms
-    def __init__(self, username, password, server, rooms=None):
+    # token    - Optional token, if set password is ignored and username is
+    #            asumed to be a user_id (@<username>:<domain>)
+    def __init__(self, username, password, server, rooms=None, token=None):
         self.username = username
 
         # Authenticate with given credentials
-        self.client = MatrixClient(server)
-        try:
-            self.client.login_with_password(username, password)
-        except MatrixRequestError as e:
-            print(e)
-            if e.code == 403:
-                print("Bad username/password")
-        except Exception as e:
-            print("Invalid server URL")
-            traceback.print_exc()
+        self.client = MatrixClient(server, user_id=username, token=token)
+        if token is None:
+            try:
+                print('logging in with password')
```

Comment:
Did you mean to leave this in, or was it just for local debugging? We probably shouldn't have `print()`s outside of error paths (and even there it's questionable, but existing code already does it...)

Reply:
Ah, no. That should have been removed. Was added for testing purposes.

Reply:
Statement removed in 68212e50c135970d242378d3c4279a9af2af9aeb.